.gitlab-ci.yml

```diff
@@ -0,0 +1,215 @@
+  script:
+   - cat /proc/meminfo | grep Free
+   - df -H
+   - lsblk
+   - nproc
+   - ulimit -a
+   - dnf --version
+
+buildenv:fedora-rawhide:rpms:
+  # primarily to download RPMs, but alas, "download" plugin not baked into image
+  <<: *buildenv-anchor  #extends: .buildenv
+  <<: *fedora-rawhide-anchor  #extends: .fedora-rawhide
+  cache:
+    key: "fedora-rawhide-dnf"
+    paths:
+      - ".cache/fedora-rawhide/dnf/rawhide-*"
+  script:
+    - mkdir -p .cache/fedora-rawhide/dnf
+    - cd .cache/fedora-rawhide
+    - mv dnf/rawhide-* /var/cache/dnf 2>/dev/null || true
+    - dnf-custom() { timeout 192 dnf -y --noplugins --nodocs
+                                 --setopt=tsflags=test
+                                 --setopt=install_weak_deps=0
+                                 --setopt=keepcache=1
+                                 --setopt=autocheck_running_kernel=0
+                                 --disablerepo='*' --enablerepo=rawhide "$@"; };
+      dnf-custom install
+        autoconf automake pkgconf-pkg-config libtool libtool-ltdl-devel
+        make findutils
+        gcc clang compiler-rt
+        libuuid-devel libxslt-devel bzip2-devel
+        'pkgconfig(glib-2.0) >= 2.16.0' 'pkgconfig(libxml-2.0)'
+        'pkgconfig(libqb) >= 0.13' 'pkgconfig(libcpg)'
+        libxml libxslt
+        psmisc which
+        bzip2-libs glib2
+        gcovr
+    - mv /var/cache/dnf/rawhide-* dnf
+
+maint:fedora-rawhide:
+  <<: *fedora-rawhide-anchor  #extends: .fedora-rawhide
+  stage: build
+  cache:
+    key: "fedora-rawhide-dnf"
+    policy: pull
+    paths:
+      - ".cache/fedora-rawhide/dnf/rawhide-*"
+  before_script:
+    - mkdir -p .cache/fedora-rawhide/dnf
+    - pushd .cache/fedora-rawhide
+    - mv dnf/rawhide-* /var/cache/dnf 2>/dev/null || true
+    - dnf-custom() { timeout 96 dnf -y --cacheonly --noplugins --nodocs
+                                --setopt=install_weak_deps=0
+                                --setopt=autocheck_running_kernel=0
+                                --disablerepo='*' --enablerepo=rawhide "$@"; };
+      dnf-custom install
+        findutils make libxml libxslt
+    - popd
+    - ( cd xml;
+        { cat Makefile.am; printf 'hack_rng\x3a %s' '${RNG_generated}'; }
+          | make -f- top_srcdir=$(pwd)/.. top_builddir=$(pwd)/.. hack_rng )
+
+  script:
+   - echo 'looking for presence of control characters...';
+      { git ls-files
+        | xargs grep -Ensv "^([^[:cntrl:]]*|$(printf '\t'))*$"||:; } 2>/dev/null
+        | { ! grep -Ev '^Binary file' && echo 'ALL OK'; };
+
+   # ./regression.sh && ./regression.sh -B && ./regression.sh -S && {
+   - cd xml;
+     ./regression.sh && ./regression.sh -B  && {
+       schemas=; for schema in *.rng; do
+         case ${schema} in *cibtr*);; *)schemas="${schemas} ${schema}";; esac;
+       done;
+       test -s .relaxng.org/relaxng.rng 2>/dev/null
+         || curl --create-dirs -Lo .relaxng.org/relaxng.rng
+            'http://relaxng.org/relaxng.rng';
```

Comment:
This is fine for convenience, but it might be worthwhile to use gitlab cache for this

Reply:
Haven't realized, just moved over from `.travis.yml`, thanks.
Incl. logic for when cache is empty/not containing this.